src/utils/getFilteredTodos.ts

```diff
@@ -0,0 +1,39 @@
+import { Todo } from '../types/Todo';
+import { Filters } from '../types/Filters';
+
+type Params = {
+  query: string;
+  selectedFilter: Filters;
+};
+
+export const getFilteredTodos = (
+  todos: Todo[],
+  { query, selectedFilter }: Params,
+) => {
+  let filteredTodos = [...todos];
```

Comment:
There's no need to create a copy of the `todos` array here, because the `filter` method does not mutate the original array but returns a new one.